homeassistant/components/overkiz/sensor.py

```diff
@@ -0,0 +1,418 @@
+"""Support for Overkiz sensors."""
+from __future__ import annotations
+
+from pyoverkiz.enums import OverkizAttribute, OverkizState, UIWidget
+
+from homeassistant.components.overkiz import HomeAssistantOverkizData
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    ENERGY_WATT_HOUR,
+    LIGHT_LUX,
+    PERCENTAGE,
+    POWER_WATT,
+    SIGNAL_STRENGTH_DECIBELS,
+    TEMP_CELSIUS,
+    TIME_SECONDS,
+    VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+    VOLUME_LITERS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, IGNORED_OVERKIZ_DEVICES
+from .coordinator import OverkizDataUpdateCoordinator
+from .entity import OverkizDescriptiveEntity, OverkizEntity, OverkizSensorDescription
+
+SENSOR_DESCRIPTIONS: list[OverkizSensorDescription] = [
+    OverkizSensorDescription(
+        key=OverkizState.CORE_BATTERY_LEVEL,
+        name="Battery Level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_BATTERY,
+        name="Battery",
+        device_class=SensorDeviceClass.BATTERY,
```

Comment:
Missing unit of measurement. If this is not a battery level sensor in percent, please remove the device class.